core/networks.go

```diff
@@ -0,0 +1,103 @@
+package core
+
+import (
+	"github.com/prysmaticlabs/prysm/shared/timeutils"
+	"github.com/sirupsen/logrus"
+	"time"
+)
+// Network represents the network.
+type Network string
+
+// NetworkFromString returns network from the given string value
+func NetworkFromString(n string) Network {
+	switch n {
+	case string(PyrmontNetwork):
+		return PyrmontNetwork
+	case string(MainNetwork):
+		return MainNetwork
+	default:
+		logrus.WithField("network", n).Fatal("undefined network")
+		return ""
+	}
+}
+
+// ForkVersion returns the fork version of the network.
+func (n Network) ForkVersion() []byte {
+	switch n {
+	case PyrmontNetwork:
+		return []byte{0, 0, 32, 9}
+	case MainNetwork:
+		return []byte{0, 0, 0, 0}
+	default:
+		logrus.WithField("network", n).Fatal("undefined network")
+		return nil
+	}
+}
+
+// DepositContractAddress returns the deposit contract address of the network.
+func (n Network) DepositContractAddress() string {
+	switch n {
+	case PyrmontNetwork:
+		return "0x8c5fecdC472E27Bc447696F431E425D02dd46a8c"
+	case MainNetwork:
+		return "0x00000000219ab540356cBB839Cbe05303d7705Fa"
+	default:
+		logrus.WithField("network", n).Fatal("undefined network")
+		return ""
+	}
+}
+
+// ForkVersion returns the fork version of the network.
+func (n Network) FullPath(relativePath string) string {
+	return BaseEIP2334Path + relativePath
+}
+
+func (n Network) MinGenesisTime() uint64 {
+	switch n {
+	case PyrmontNetwork:
+		return 1605700800
+	case MainNetwork:
+		return 1606824000
```

Comment:
should the genesis time reflect the delay? 23 sec for mainnet and 7 for pyrmont

Reply:
which delay you mean? I took those numbers from prysm

Reply:
I mean that MainNet Genesis is at 2020-12-01 12:00:23 UTC and not at 2020-12-01 12:00:00 UTC as calculated from 1606824000

Reply:
fixed for both